[WIP] systemd integration draft. (#34)
emqx/eip: active/0012-systemd-integration.md

# Integration with Systemd

## Changelog

* 2021-06-09: @zhongwencool Initial draft

## Abstract

> Emqx currently uses a simple systemd forking type to start emqx in demon mode. The other benefits of systemd are not fully exploited, for example, systemd does not restart all emqx processes that exit abnormally (`kill GroupChildrenPid`). Systemd  just acts as a starter.

Reviewer: daemon 
```suggestion
emqx currently uses a simple systemd forking type to start emqx in daemon mode. Systemd is only acting as a program starter, benefits of it are not fully exploited. For example, systemd does not restart all emqx processes that exit abnormally (`kill GroupChildrenPid`).
```

> 

Reviewer: Maybe add a summary of what is the goal of this proposal.
e.g.
```
The document is to share the idea about how we can make emqx better integrated with systemd
```

## Motivation

The benefits of [systemd](https://github.com/systemd/systemd) include, but are not limited to:

> - The ability to configure specific system parameters for service such as `LimitNOFILE`.

Reviewer: This is already the case even in fork mode ?
if yes, maybe better to state it here.

- The reboot interval can be configured, and restart after the machine reboots.

> - Journalctl logging can be configured.

Reviewer: need more elaboration on this,
doe fork mode limit us from utilising journal logs ?

- Very stable, easy and configurable way to manage services.

> This will make the node more robust and the user will see more information about the operation of the node via `systemctl status emqx`, Make `start/stop/restart` work as expected.

Reviewer: need more elaboration on this too about: how is `systemct` broken (or note good enough) today.
